### ISSUE TITLE
Warning: config value 'plugins': Plugin 'exporter' option 'formats': Unrecognised configuration name: formats

### ISSUE THREAD

**Issue by stephffuller** [adrienbrignon/mkdocs-exporter#32]
I know this is probably something stupid like a missing space somewhere, but I've spent a couple of days banging my head against this and I cannot get this plugin to generate a pdf. I believe the error in the title above might be the culprit. I copied my config file directly from the docs for the plugin, and only changed the filenames and locations for the cover page and the output file. Please have a look and tell me what I'm doing wrong. 

![image](https://github.com/user-attachments/assets/ec4983de-4427-4797-9286-21bdd5fd0b1f)

**Comment by stephffuller** [comment]
@adrienbrignon any thoughts? Sorry to bug, but this is blocking me. I really need this plugin to work. Many thanks in advance for any suggestions.

**Comment by adrienbrignon** [comment]
Hello @stephffuller,

Do you have a repository to share in order to reproduce the issue, please?
In the meantime, I'll investigate.

Thanks,

**Comment by stephffuller** [comment]
Sadly, our repo is private so I cannot share. But I'm willing to make any changes you suggest and send the output. 
![image](https://github.com/user-attachments/assets/42fdbf00-ae25-4d0c-b34f-61502dc7db7c)

**Comment by adrienbrignon** [comment]
Thank you for the feedback @stephffuller.
I understand, no problem.

You are using the version 5 of the plugin, which is now outdated.
I guess that you could upgrade to the latest version ([`6.1.1`](https://github.com/adrienbrignon/mkdocs-exporter/releases/tag/v6.1.1)) without any issue, that should solve the problem you are facing.

**Comment by stephffuller** [comment]
Ok, upgrading got me some movement on this, but alas, no pdf yet. But at least now it is actually trying to generate one! I had to add download buttons, as apparently they're required now? It wouldn't execute the mkdocs build without them.

It seems like it's having trouble with links in the docs? All the links work in the html site, and mkdocs itself doesn't report any broken links, so I'm not sure why mkdocs-exporter is giving me hundreds of these, all for different files: 

![image](https://github.com/user-attachments/assets/143362ef-4649-424d-974f-157fcc2f20ac)

And then it errors out with this: 
![image](https://github.com/user-attachments/assets/015ad182-b065-40fa-8862-3269a5ebc5af)

![image](https://github.com/user-attachments/assets/8c366340-a8ef-443c-ad4b-c96e6b1e51df)

Also, I love your Rick Roll on the download buttons page *chef's kiss*. 

I'll contribute some docs improvement PRs once I get this working as appreciation for the help!

**Comment by adrienbrignon** [comment]
Nice catch about the buttons, I accidentally made that option mandatory...
As a workaround, you can define the `buttons` option as an empty array, until I fix this issue in the next patch release.

Regarding the errors you are facing, it seems that there is something on your pages (maybe some JavaScript?) that is blocking the rendering, hence the timeout being reached.

You can try to disable the browser's headless mode (with the [`browser.headless` option](https://adrienbrignon.github.io/mkdocs-exporter/reference/configuration/formats/pdf/#mkdocs_exporter.formats.pdf.config.BrowserConfig.headless) set to `false`) and inspect the browser's console for logs.

**Comment by stephffuller** [comment]
Ok, I may know what is happening then. We have links to video tutorials hosted on Wistia in our docs. The plugin we were trying to use for PDFs, mkdocs-print-site-plugin, had an option to ignore those sections with the videos so that the pdf didn't contain those. You can see the options [here](https://timvink.github.io/mkdocs-print-site-plugin/how-to/do_not_print.html#ignoring-elements-in-a-page). Does mkdocs-exporter have a similar option to ignore just a section or part of a page? I didn't see one. I can do a global find/replace if one does exist and it will probably fix this issue.

**Comment by adrienbrignon** [comment]
Glad you found the root of the issue!

The plugin includes a similar feature that [removes any HTML elements](https://github.com/adrienbrignon/mkdocs-exporter/blob/f78ad8e594bfd44124d67547d3f50a86ff10b921/mkdocs_exporter/plugin.py#L86) with the `data-decompose` attribute set to `true` (e.g., `<div data-decompose="true">bye</div>`). This feature is currently undocumented, as it is mainly used internally to remove cover pages once they've been rendered. However, you should be able to leverage it.

In the next release, I'll introduce a documented feature that offers similar functionality using a CSS class, such as `mkdocs-exporter-ignore` :)

**Comment by stephffuller** [comment]
@adrienbrignon so would I just add 

<div class="mkdocs-exporter-ignore">
  <p>I wont appear in the document</p>
</div>

into my markdown files around those videos? Thanks so much for your help with this.

**Comment by adrienbrignon** [comment]
Yes, this should work as expected once this feature is shipped within the next release.
Any DOM element with the `mkdocs-exporter-ignore` class will be removed before printing the page.

**Comment by stephffuller** [comment]
Hi @adrienbrignon . Wanted to let you know where we are. I was able to configure to ignore the videos, but ran into another similar issue, and these I cannot ignore for the output. 

We are using the [`extra` config option for mkdocs](https://www.mkdocs.org/user-guide/configuration/#extra) that allows us to set variables that are then replaced in the final output. To reference those variables, in the markdown files we use a syntax similar to what was used for the videos that mkdocs-exporter had a problem with:

```
...the upgrade process for {{ SAAP }} ...
```

So now it looks like mkdocs-exporter is not able to produce the pdf with those variables in place. Could you add support for the `extra` config in a future release? Many thanks.
